homeassistant/components/dlink/switch.py

```diff
@@ -47,10 +47,7 @@
     }
 )
 
-SWITCH_TYPE = SwitchEntityDescription(
-    key="switch",
-    name="Switch",
-)
+SWITCH_TYPE = SwitchEntityDescription(key="")
```

Comment:
Changing the key will change the unique id. We can't do that without a migration. I suggest not changing it.

Reply:
What unique id? From https://github.com/home-assistant/core/pull/84927/files, only the name was set before the config flow. No unique id was set before that. We import the name as the entry title which becomes the entity id for the switch.

Reply:
I will however wait until removing the import to reintroduce the change if you'd like.

Reply:
We set a unique id here:

https://github.com/home-assistant/core/blob/3b5fd4bd067949fef1263f709f3b6eca2d0b3aef/homeassistant/components/dlink/entity.py#L31

Reply:
Yes of course but the config flow is only in beta if that matters so people running 2023.1 would not see a breaking change.

Reply:
You mean you want to release this change in 2023.2.0?

I don't really see the point of changing the key here. It's not a bug fix, right?

Reply:
No. Would not qualify for that. So then just the name will be changed with import removal.